### DISCUSSION TITLE
Option to Ignore certain characters in search (e.g., dashes)

### DISCUSSION THREAD

**Discussion by nicolnt** [tadashi-aikawa/obsidian-another-quick-switcher#249] [Q&A] [answered]
Hi,

I find the search quite restrictive, even in fuzzy mode, regarding the presence or absence of certain characters, especially dashes “-”.
Is there a way to ignore those if they are present in the note title or alias and omitted in the search?

## Example case
Here's an example with a note titled or aliased as “Bio-based products LCAs”:

### Search Results:
- “biobased” gives no results
- “:ff biobased products lcas” gives no results as well
- “:ff biobased” gives unrelated results

Thanks for considering this!

**Comment by tadashi-aikawa** [comment]
Hi, @nicolnt :)

Unfortunately, I was able to hit the note in my environment.

![image](https://github.com/user-attachments/assets/a36869cc-7332-467c-8649-3c88aebb17db)

Could you either provide the reproduction steps in the Sandbox Vault or share the search settings you used from `${your_vault}/.obsidian/plugins/obsidian-another-quick-switcher/data.json` ?

ex: The case you use "Recent search"
![image](https://github.com/user-attachments/assets/a4d73e85-7b23-476f-9da0-8a4b7525fb22)

**Reply by tadashi-aikawa** [reply]
@nicolnt 
Another common case is that the current file does not appear in the suggestions. Just in case, I wanted to mention this as an additional note.

**Reply by nicolnt** [reply]
Hi @tadashi-aikawa 

I could reproduce it in Sandbox vault.

- Just created a note titled "Bio-based"
- Install "Another Quick Switcher" Version: 11.5.1
- Enable "Another Quick Switcher" Version: 11.5.1
- Trigger "Another Quick Switcher: Recent search" from the command palette
- Type "Biobased"

![image](https://github.com/user-attachments/assets/c89ff47e-9243-4bfb-95df-8175386dde11)

Below with Obsidian's default quick switcher

![image](https://github.com/user-attachments/assets/524c93f2-3785-4caf-af32-c1898d2c134f)

<details>
  <summary>Debug info (sandbox vault)</summary>

```
  SYSTEM INFO:
	Obsidian version: v1.6.7
	Installer version: v1.5.3
	Operating system: Windows 10 Pro 10.0.19045
	Login status: logged in
	Catalyst license: insider
	Insider build toggle: off
	Live preview: on
	Base theme: adapt to system
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 1
	Plugins enabled: 1
		1: Another Quick Switcher v11.5.1

RECOMMENDATIONS:
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.
```


</details>

**Reply by tadashi-aikawa** [reply, marked as answer]
Thank you, @nicolnt :)  I could reproduce it 👍 

Perhaps, you should enable the Allow fuzzy search for "Search target" option. (default: disabled)

![image](https://github.com/user-attachments/assets/e68adb9c-0442-4b41-a402-69d2ff5b9f14)

**Reply by nicolnt** [reply]
Hi @tadashi-aikawa 
I submitted a new issue because this option doesn't stay on https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/issues/252

**Reply by tadashi-aikawa** [reply]
@nicolnt 
I just replied to the issue.
